packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js

```diff
@@ -223,11 +232,31 @@ function FontCollection( { slug } ) {
 		);
 	}
 
+	let actionsComponent = null;
+	if ( ! renderConfirmDialog && ! selectedFont ) {
```

Comment:
This should include checking whether the font collection rendered is the Google Fonts one. Otherwise, the component also renders on unwanted font collections (not the Google fonts one).

For example 
```js
slug === "google-fonts"
```

How it currently looks:
![image](https://github.com/WordPress/gutenberg/assets/1310626/70f117b8-e7b0-418c-9304-2f8913447640)